server/lib/tuist_web/components/layouts/app.html.heex

```diff
@@ -1,10 +1,31 @@
 <!DOCTYPE html>
-<html lang="en" class="[scrollbar-gutter:stable]" data-theme="light">
+<html lang="en" class="[scrollbar-gutter:stable]">
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="csp-nonce" content={get_csp_nonce()} />
     <meta name="csrf-token" content={get_csrf_token()} />
+    <script nonce={get_csp_nonce()}>
+      (function() {
+        function getPreferredTheme() {
+          try {
+            const preferredTheme = localStorage.getItem("preferred-theme");
+            return preferredTheme === "null" || preferredTheme === null ? "system" : preferredTheme;
+          } catch (e) {
+            return "system";
+          }
+        }
+        
+        function setInitialTheme() {
+          const theme = getPreferredTheme();
+          const resolvedColorScheme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
+          document.documentElement.style.setProperty("color-scheme", theme === "system" ? "light dark" : theme);
+          document.documentElement.setAttribute("data-theme", theme === "system" ? resolvedColorScheme : theme);
+        }
+        
+        setInitialTheme();
+      })();
+    </script>
```

Comment:
I'd extract this in a component that we can reuse for the marketing page too.

Reply:
correct me if I'm wrong, but we're not using dark mode for the marketing page. And we certainly shouldn't support overriding the theme there (and so shouldn't depend on `data-theme` being set). So, I don't think that's needed.